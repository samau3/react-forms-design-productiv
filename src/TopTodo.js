import React from "react";

import Todo from "./Todo";

/** Shows the top todo.
 *
 * Props:
 * - todos
 *
 * TodoApp -> TopTodo
 */

function TopTodo({ todos }) {
  // lowest-priority # is the highest priority
  // console.log("TopTodo todos", todos)
  let top = todos.reduce(
    (acc, cur) => cur.priority < acc.priority ? cur : acc, todos[0]);

  const { id, title, description, priority } = top;
  console.log("TopTodo:", id, title, description, priority);

  return <Todo
    id={id}
    title={title}
    description={description}
    priority={priority}
  />;
}

export default TopTodo;