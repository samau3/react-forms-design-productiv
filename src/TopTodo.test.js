import React from "react";
import { render } from "@testing-library/react";

import TEST_DATA from "./_testCommon";
import TopTodo from "./TopTodo";


// const { id, title, description, priority } = TEST_DATA[0];


it("renders without crashing", function () {
    render(<TopTodo todos={TEST_DATA} />);
});

it("matches snapshot", function () {
    console.log(TEST_DATA)
    const { asFragment } = render(<TopTodo todos={TEST_DATA} />);
    expect(asFragment()).toMatchSnapshot();
});

it("contains TopTodo todo information", function () {
    const { container, debug } = render(<TopTodo todos={TEST_DATA} />);
    debug(container)
    expect(container.querySelector('.Todo'))
        .toBeInTheDocument();
});

// other tests for TopTodo:
// could also see what zero, one renders