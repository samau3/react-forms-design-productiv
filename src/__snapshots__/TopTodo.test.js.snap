// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <div>
      <b>
        Make dinner
      </b>
       
      <small>
        (priority: 1)
      </small>
    </div>
    <div>
      <small>
        Cook something healthy
      </small>
    </div>
  </div>
</DocumentFragment>
`;
